### PR TITLE
Bunch of fixes and improvements

### PR DESCRIPTION
**Pull request by kasperpeulen**
* Use MSW to mock OAuth
* use Redirect boundary for invalidation
* use standard autoincrementing ids 
* Fix some type issues
* Add nextjs.navigation parameters to all pages
* Refactor auth callback to be a regular endpoint, which is easier to mock
* Delete all other middleware as it was unused
* Make search a regular action, to use latest next standards

### CODE REVIEW

File: .env.local.example
Comment (JReinhold) [inline .env.local.example:1]: This is mentioned in the Readme, so you need to update that as well.

File: app/note/[id]/page.stories.tsx [hunk 1]
```diff
@@ -45,14 +53,69 @@ export const LoggedIn: Story = {
...
+  },
   beforeEach() {
-    useSearchParams.mockReturnValue({ get: () => 'RSC' })
+    // Point the login implementation to the endpoint github would have redirected too.
+    login.mockImplementation(async () => {
+      return await auth.GET(new Request('/auth?code=storybookjs'))
+    })
+  },
+  play: async ({ canvasElement }) => {
+    console.log(db.$getInternalState())
```
Comment (JReinhold) [inline app/note/[id]/page.stories.tsx:83]: remove?

**Review by JReinhold — APPROVED** [review]
LGTM! ✨ You also tested that the application still works right?

File: .env.local.example
Comment (kasperpeulen) [reply .env.local.example:1]: actually not sure why I deleted that

File: app/note/[id]/page.stories.tsx (hunk 1 above)
Comment (kasperpeulen) [reply app/note/[id]/page.stories.tsx]: Done

**Comment by kasperpeulen** [comment]
Yes, application works!

**Pull request merged**